web/app/Providers/AppServiceProvider.php

```
@@ -14,6 +16,28 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
+        // we don't build the client assets in the php-fpm container,
+        // so we'll need to retrieve the manifest for them from the
+        // nginx container
+
+        $vite_manifest_path = config('vite.build_path') . '/manifest.json';
+
+        if (!file_exists(public_path($vite_manifest_path))) {
+            // get manifest from nginx container
+            // we have to catch a potential exception here, because
+            // nginx may or may not be running.
+            // also, if the manifest isn't there laravel-vite will just
+            // throw its own error, so we'll just let it handle that
+            try {
+                $file = file_get_contents("http://nginx:80/$vite_manifest_path");
+
+                if ($file) {
+                    file_put_contents(public_path($vite_manifest_path), $file);
+                    Log::debug('Wrote Vite manifest file (retrieved from nginx)');
+                }
+            } catch (Exception $err) {
+            }
```

Why is there an empty catch here? Either the error should surface, or we should log if we're fine this not succeeding.

---

Keep in mind that this gets ran on every request - so if it fails, it'll just get tried again. Additionally, if the manifest really just isn't there, or can't be written, `laravel-vite` will throw its own errors. So, basically, I _am_ surfacing the error.

---

In that case please leave a comment. I'm not comfortable with empty catch blocks since it's hard to tell what's going on there.

---

But... I did? There is a big comment above the `try` block.

---

Oh I see what you mean